feat: add apple privacy manifest (#276)
amplitude/amplitude-dev-center: docs/data/sdks/ios-swift/index.md

If you have any problems or issues with the SDK, [create a GitHub issue](https://github.com/amplitude/Amplitude-Swift/issues/new) or submit a request on [Amplitude Help](https://help.amplitude.com/hc/en-us/requests/new).

--8<-- "includes/abbreviations.md"

### Apple privacy manifest

> Starting December 8, 2020, Apple requires a privacy manifest file for all new apps and app updates and Apple expects to make it mandatory in the Spring of 2024. As Amplitude is a third-party to your app, you’ll need to ensure you are properly disclosing to your users the ways you are using Amplitude in regards to their data.

Reviewer: dec 8 2020?

Author: Yes. https://developer.apple.com/news/?id=em8fm29e

> Starting December 8, 2020, Apple requires a privacy manifest file for all new apps and app updates and Apple expects to make it mandatory in the Spring of 2024. As Amplitude is a third-party to your app, you’ll need to ensure you are properly disclosing to your users the ways you are using Amplitude in regards to their data.

Reviewer 2: ```suggestion
Starting December 8, 2020, Apple requires a privacy manifest file for all new apps and app updates. Apple expects to make this mandatory in the Spring of 2024. As Amplitude is a third-party to your app, you need to ensure you properly disclose to your users the ways you use Amplitude in regards to their data.
```

Author: Thanks @markzegarelli. I will avoid using future tense.

!!! Note "Update privacy manifest based on your app"

>     Amplitude sets privacy manifest based on the default configuration. Update the privacy manifest according to your configuraiton and your app.

Reviewer 2: ```suggestion
    Amplitude sets privacy manifest based on a default configuration. Update the privacy manifest according to your configuration and your app.
```

#### NSPrivacyTracking

!!! Info "Tracking definition"

>     Tracking refers to the act of linking user or device data collected from your app with user or device data collected from other companies' apps, websites, or offline properties for targeted advertising or advertising measurement purposes. Read more [here](https://developer.apple.com/app-store/user-privacy-and-data-use/).

Reviewer 2: ```suggestion
    Tracking refers to the act of linking user or device data collected from your app with user or device data collected from other companies' apps, websites, or offline properties for targeted advertising or advertising measurement purposes. For more information, see Apple's article [User privacy and data use](https://developer.apple.com/app-store/user-privacy-and-data-use/).
```

> Amplitude by default doesn't use data for tracking. Add this field and set it to true if your app does.

Reviewer 2: ```suggestion
By default, Amplitude doesn't use data for tracking. Add this field and set it to true if your app does.
```

#### NSPrivacyCollectedDataTypes

| Date type | Linked to user | Used for tracking | Reason for collection |
| --------- | -------------- | ------------ | --------------------- |
|Product interaction|Yes|No|Analytics|
|Device ID|Yes|No|Analytics|

> |Coarse Location|Yes|No|Analytics|

Reviewer 3: we may want to call out this is based on ip geolocation, and not something we collect from device gps/location features.

Author: I added a column to explain where each filed is tracked at https://github.com/amplitude/amplitude-dev-center/pull/276/commits/917fd497a7804eb9eb357885cc221433f9a2f70f
![image](https://github.com/amplitude/amplitude-dev-center/assets/31029607/6b16a414-e44e-4c11-94f7-5dcf761d958e)

> By default the SDK tracks device Id only, but you can track user Id as well by `amplitude.setUserId()`. To do so, you have to add "User ID" Data type. See all data types [here](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests).

Reviewer 2: ```suggestion
By default the SDK tracks `deviceId` only. You can use `amplitude.setUserId()` track `userId` as well. To do so, add the "User ID" Data type. For more information about data types, see Apple's article [Describing data use in privacy manifests](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests).
```

> By default the SDK tracks `deviceId` only. You can use `amplitude.setUserId()` track `userId` as well. To do so, add the "User ID" Data type. For more information about data types, see Apple's article [Describing data use in privacy manifests](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests).

Reviewer: You can use `amplitude.setUserId()` *to* track ...

> #### NSPrivacyTrackingDomains

Reviewer 3: I'm not sure if we want to include these by default - Consider an app that is tracking users, but not through Amplitude, instead just relying on us for analytics. The [docs](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files) imply that requests to our domains would be blocked if the user requests not to be tracked. Perhaps we add these as a suggestion on the docs for customers that are actually tracking through us?

Author: > If you set NSPrivacyTracking to true then you need to provide at least one internet domain in NSPrivacyTrackingDomains; otherwise, you can provide zero or more domains.

It makes sense to not have tracking domains by default. There is PR to update the privacy manifest https://github.com/amplitude/Amplitude-Swift/pull/155

Reviewer 3: Thanks!

Author: Updated docs at https://github.com/amplitude/amplitude-dev-center/pull/276/commits/a6d068b405bdb6f9af830ea6ce56e26a1f98a936
